Source/ablastr/utils/SignalHandling.cpp

```diff
@@ -110,6 +110,18 @@ SignalHandling::parseSignalNameToNumber(const std::string &str)
     return sig;
 }
 
+bool
+SignalHandling::anySignalConfActive ()
+{
+    bool any_signal_active = false;
```

Comment:
Should this maybe be a member variable of `SignalHandling`, that can be set just once, rather than looping over everything twice every step?

Reply:
that probably makes sense, yes.
I was coding this in a way  that one could register signals later on, e.g., from the Python layer. But not sure if that's truly needed.

We set that variable at the end of the config step, so even if someone configures it later, it would be set consistently.